public/app/features/alerting/unified/components/MenuItemPauseRule.tsx

```diff
@@ -24,7 +25,7 @@ interface Props {
  */
 const MenuItemPauseRule = ({ rule, onPauseChange }: Props) => {
   const notifyApp = useAppNotification();
-  const [pauseRule, updateState] = usePauseRuleInGroup();
+  const [updateState, pauseRule] = usePauseRuleInGroup();
```

Comment:
It feels a little strange/unfortunate that this is the other way round from RTKQ mutations, but I guess it's best to keep the signature of the copied useAsync hook 🤷

Reply:
yea there's the trade-off here between developer experience in its consumption and how the hooks are implemented; rather than de-structuring the `useAsync` array result and reconstructing it in the hooks implementation I've chosen to just keep whatever the `useAsyncFn` hook chose.

TypeScript _should_ help here though in case we try to destruct the fn or vice versa :D We can always change our minds about this later if we feel this is painful.